core/common/token-core/src/test/java/org/eclipse/edc/token/JwtGenerationServiceTest.java

```diff
@@ -88,13 +80,29 @@ void verifyTokenGeneration() throws ParseException, JOSEException {
         });
     }
 
+    @Test
+    void shouldFail_whenPrivateKeyCannotBeResolved() {
+        var decorator = testDecorator();
+
+        var result = tokenGenerationService.generate(() -> null, decorator);
+
+        assertThat(result.failed()).isTrue();
+    }
+
     private JWSVerifier createVerifier(JWSHeader header, Key publicKey) throws JOSEException {
         return new DefaultJWSVerifierFactory().createJWSVerifier(header, publicKey);
     }
 
+    private RSAKey testKey() throws JOSEException {
```

Comment:
suggestion: I would use Ed25519 keys wherever possible, as they seem to emerge as the new standard. also, they are much faster to generate than RSA keys: 
with Nimbus (this is the one we'd need here): 
```java
JWK ed25519Key = new OctetKeyPairGenerator(Curve.Ed25519).generate();
```
with plain Java:
```java
KeyPair kp = KeyPairGenerator.getInstance("Ed25519").generateKeyPair();
```

using the BouncyCastle provider:
```java
KeyPair kp = KeyPairGenerator.getInstance("Ed25519", new BouncyCastleProvider()).generateKeyPair();
```

Reply:
with the first one I get a `Export to java.security.PrivateKey not supported`, the second one tells me that the algorithm is not supported... 
to be honest I didn't change the code here, just moved the method at the bottom, if you're ok with it I would leave it as it is

Reply:
yeah it's fine. we can do that in a housekeeping task somewhere down the road.